[TypeCheckAttr] @frozen is valid even when building without enable-library-evolution

nate-chandler opened the pull request:
Even when building not for library evolution, `@frozen` is a valid attribute on enums.

Fixes an issue where conformances to `Sendable` and `BitwiseCopyable` of `@frozen` enums are not inferred in targets not built for library evolution such as the embedded stdlib.

rdar://128358780

nate-chandler commented:
@swift-ci please test

nate-chandler commented:
@swift-ci please test

nate-chandler commented:
Alternatively
```
diff --git a/lib/Sema/TypeCheckAttr.cpp b/lib/Sema/TypeCheckAttr.cpp
index efe408787f6..75d2bfd4e47 100644
--- a/lib/Sema/TypeCheckAttr.cpp
+++ b/lib/Sema/TypeCheckAttr.cpp
@@ -3960,11 +3960,6 @@ void AttributeChecker::visitImplementsAttr(ImplementsAttr *attr) {

 void AttributeChecker::visitFrozenAttr(FrozenAttr *attr) {
   if (auto *ED = dyn_cast<EnumDecl>(D)) {
-    if (!ED->getModuleContext()->isResilient()) {
-      attr->setInvalid();
-      return;
-    }
-
     if (ED->getFormalAccess() < AccessLevel::Package &&
         !ED->getAttrs().hasAttribute<UsableFromInlineAttr>()) {
       diagnoseAndRemoveAttr(attr, diag::enum_frozen_nonpublic, attr);
```

nate-chandler commented:
@swift-ci please test

nate-chandler commented:
@swift-ci please test windows platform

DougGregor commented:
> Alternatively

This is the way! The attribute checker shouldn't be marking this attribute invalid.

nate-chandler committed:
[TypeCheckAttr] Frozen is valid without resilience

Even when building not for library evolution, @frozen is a valid
attribute on enums.

rdar://128358780

nate-chandler commented:
@swift-ci please test

nate-chandler commented:
@swift-ci please test source compatibility

nate-chandler commented:
@swift-ci please test

nate-chandler commented:
@swift-ci please test source compatibility

nate-chandler commented:
@swift-ci Please Test Source Compatibility Debug
